File: windows/src/Aegis.Windows.App/Diagnostics/Diagnostics.cs
namespace Aegis.Windows.Diagnostics;

using System;
using System.Collections.Generic;

/// <summary>阶段 C（蓝图 windows/src/Aegis.Windows.Diagnostics）：
/// crash/health 非敏感日志骨架——脱敏、速率限制、不可把 token/网页内容
/// 写入日志（蓝图迁移表——event_log/crash_reporter 迁入）。阶段 C 最小。</summary>
public sealed class Diagnostics
{
    private readonly List<string> _nonSensitiveLog = new();

    public void LogNonSensitive(string message)
    {
        // 只记录非敏感信息（不包含 token/网页内容/query secret——脱敏）
        _nonSensitiveLog.Add($"[{DateTime.UtcNow:o}] {message}");
    }

    public IReadOnlyList<string> Log => _nonSensitiveLog;
}

File: android/diagnostics/src/main/kotlin/com/aegis/diagnostics/Diagnostics.kt
package com.aegis.diagnostics

/**
 * 阶段 D（蓝图 android/diagnostics）：脱敏诊断日志骨架——不记录 token/网页内容/
 * query secret（蓝图迁移表：event_log/crash_reporter 迁入——脱敏、速率限制）。
 * 阶段 D 最小——非敏感健康/崩溃信息（完整实现按蓝图迭代）。
 */
class Diagnostics {
    private val log = mutableListOf<String>()

    fun logNonSensitive(message: String) {
        // 只记录非敏感信息（不包含 token/网页内容/query secret——脱敏）
        log += "${java.time.Instant.now()} $message"
    }

    val entries: List<String> get() = log
}
